src/webui/www/private/scripts/torrent-content.js

```diff
@@ -0,0 +1,572 @@
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+"use strict";
+
+window.qBittorrent ??= {};
+window.qBittorrent.TorrentContent ??= (() => {
+    const exports = () => {
+        return {
+            init: init,
+            normalizePriority: normalizePriority,
+            isFolder: isFolder,
+            createDownloadCheckbox: createDownloadCheckbox,
+            updateDownloadCheckbox: updateDownloadCheckbox,
+            createPriorityCombo: createPriorityCombo,
+            updatePriorityCombo: updatePriorityCombo,
+            updateData: updateData,
+            clearFilterInputTimer: clearFilterInputTimer
+        };
+    };
+
+    let torrentFilesTable;
+    const FilePriority = window.qBittorrent.FileTree.FilePriority;
+    const TriState = window.qBittorrent.FileTree.TriState;
+    let torrentFilesFilterInputTimer = -1;
+    let onFilePriorityChanged = null;
+
+    const normalizePriority = (priority) => {
+        priority = Number(priority);
+
+        switch (priority) {
+            case FilePriority.Ignored:
+            case FilePriority.Normal:
+            case FilePriority.High:
+            case FilePriority.Maximum:
+            case FilePriority.Mixed:
+                return priority;
+            default:
+                return FilePriority.Normal;
+        }
+    };
+
+    const triStateFromPriority = (priority) => {
+        switch (normalizePriority(priority)) {
+            case FilePriority.Ignored:
+                return TriState.Unchecked;
+            case FilePriority.Normal:
+            case FilePriority.High:
+            case FilePriority.Maximum:
+                return TriState.Checked;
+            case FilePriority.Mixed:
+                return TriState.Partial;
+        }
+    };
+
+    const isFolder = (fileId) => {
+        return fileId === -1;
+    };
+
+    const getAllChildren = (id, fileId) => {
+        const getChildFiles = (node) => {
+            rowIds.push(node.data.rowId);
+            fileIds.push(node.data.fileId);
+
+            if (node.isFolder) {
+                node.children.forEach((child) => {
+                    getChildFiles(child);
+                });
+            }
+        };
+
+        const node = torrentFilesTable.getNode(id);
+        const rowIds = [node.data.rowId];
+        const fileIds = [node.data.fileId];
+
+        node.children.forEach((child) => {
+            getChildFiles(child);
+        });
```

Comment:
The old code is really poorly written. Instead of letting `getChildFiles` take a root node and do all operations within it, it duplicates code unnecessarily to the outside.

Reply:
This PR doesn't make it worse, so I think improving the existing implementation is best left for a future PR.

Reply:
don't mind that, it was just some mumbling...